py-polars/src/polars/expr/string.py

```diff
@@ -2709,6 +2708,66 @@ def replace_many(
         │ Tell me what you want, what you really really want ┆ Tell you what me need, what me really really need │
         │ Can you feel the love tonight                      ┆ Can me feel the love tonight                      │
         └────────────────────────────────────────────────────┴───────────────────────────────────────────────────┘
+
+        Using `leftmost` and changing order of tokens in `patterns`, you can get fine control over replacement
+        logic, following `Aho-Corasick crate documentation <https://docs.rs/aho-corasick/latest/aho_corasick/struct.AhoCorasickBuilder.html#method.match_kind>`_
+        and its examples.
```

Comment:
I don't want to explicitly mention our internal implementation, or codify whatever behavior it has for when `leftmost` is not set. The only thing I'd like our documentation to mention is that when `leftmost` is not set there are no guarantees in which order overlapping matches are considered or replaced.